### DISCUSSION TITLE
Part 3(Foundry Fundamentals - Lesson 19 - Anvil not running on VSC

### DISCUSSION THREAD

**Discussion by ShawnFromTexas** [Cyfrin/foundry-full-course-cu#1265] [Q&A] [answered]
After working on ThirdWeb with SimpleStorage contract my anvil wont run on WSCode. Getting this message when trying to run again for a new terminal.

```solidity
error binding to 127.0.0.1:8545: error creating server listener: Address already in use (os error 98)
```

I have restarted WSCode and disconnected from ThirdWeb thinking it may be that. Also logged out of MetaMask just in case but no luck. any help much appreciated!

**Comment by ShawnFromTexas** [comment, marked as answer]
SOLVED! I had to run the following command in my terminal to identify the process using port 8545: then needed to TERMINATE CONFLICTING PORT OR PROCESS. Follow code below 

```solidity
lsof -i :8545
```

This will show you the process ID (PID) of the application using port 8545.
Terminate the Conflicting Process:

If the process using port 8545 is not essential, you can terminate it. You can use the following command to kill the process:

```solidity
kill -9 <PID>
```

Replace <PID> with the actual process ID you obtained from the previous step. Once finished run code below again to check port is killed properly. 

```solidity
lsof -i :8545
```

Runnig the lsof -i :8545 command again will ensure that the port is no longer in use. If the process was successfully terminated, there should be no output.
Run Anvil Application:

Try running your Anvil application again and see if the issue is resolved. cheers!

**Reply by xyz899** [reply]
Hi,

Since you found it, mark it as solved then

**Reply by ShawnFromTexas** [reply]
Yes i have been trying to find that tab. Can you help fren? insaneeee!!!

**Reply by ShawnFromTexas** [reply]
Solved! Many thanks for letting me know and getting me to figure this out! cheers!
